cmd/gazelle/diff.go

```diff
@@ -44,6 +44,8 @@ func diffFile(c *config.Config, f *rule.File) error {
 
 	if len(f.Content) == 0 {
 		diff.FromFile = "/dev/null"
+		// difflib.SplitLines will turn the empty string into "\n"
+		diff.A = "\n"
```

Comment:
Seems like the main issue here is that `len(f.Content) == 0` regardless of whether the file was empty or non-existent. It would be better to pass in more information to distinguish those two cases.

If the file doesn't exist, and `f.Content` is empty after formatting, then we don't have to print anything (here or in `printFile`).

Reply:
Could use `[]byte(nil)` vs. `[]byte{}` to differentiate

Reply:
So I updated it to check if the file was explicitly non-existent for now, but I'm open to refining it further.